codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -0,0 +1,642 @@
+import software.amazon.smithy.rust.codegen.rustlang.rustBlock
+import software.amazon.smithy.rust.codegen.rustlang.rustBlockTemplate
+import software.amazon.smithy.rust.codegen.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.rustlang.withBlock
+import software.amazon.smithy.rust.codegen.rustlang.writable
+import software.amazon.smithy.rust.codegen.server.smithy.generators.protocol.ServerProtocolGenerator
+import software.amazon.smithy.rust.codegen.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.smithy.RuntimeConfig
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.generators.StructureGenerator
+import software.amazon.smithy.rust.codegen.smithy.generators.builderSymbol
+import software.amazon.smithy.rust.codegen.smithy.generators.error.errorSymbol
+import software.amazon.smithy.rust.codegen.smithy.generators.http.ResponseBindingGenerator
+import software.amazon.smithy.rust.codegen.smithy.generators.protocol.MakeOperationGenerator
+import software.amazon.smithy.rust.codegen.smithy.generators.protocol.ProtocolTraitImplGenerator
+import software.amazon.smithy.rust.codegen.smithy.generators.setterName
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpBindingDescriptor
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpBoundProtocolBodyGenerator
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpBoundProtocolTraitImplGenerator
+import software.amazon.smithy.rust.codegen.smithy.protocols.HttpLocation
+import software.amazon.smithy.rust.codegen.smithy.protocols.Protocol
+import software.amazon.smithy.rust.codegen.smithy.transformers.errorMessageMember
+import software.amazon.smithy.rust.codegen.util.dq
+import software.amazon.smithy.rust.codegen.util.expectTrait
+import software.amazon.smithy.rust.codegen.util.getTrait
+import software.amazon.smithy.rust.codegen.util.hasStreamingMember
+import software.amazon.smithy.rust.codegen.util.inputShape
+import software.amazon.smithy.rust.codegen.util.outputShape
+import software.amazon.smithy.rust.codegen.util.toSnakeCase
+import java.util.logging.Logger
+
+/*
+ * Implement operations input parsing and output serialization. Protocols can plug their own implementations
+ * and overrides by creating a protocol factory inheriting from this class and feeding it to the ServerProtocolLoader.
+ * See ServerRestJson.kt for more info.
+ */
+class ServerHttpProtocolGenerator(
+    codegenContext: CodegenContext,
+    protocol: Protocol,
+) : ServerProtocolGenerator(
+    codegenContext,
+    protocol,
+    MakeOperationGenerator(codegenContext, protocol, HttpBoundProtocolBodyGenerator(codegenContext, protocol)),
+    ServerHttpProtocolImplGenerator(codegenContext, protocol),
+)
+
+/*
+ * Class used to expose Rust server traits types. Is is used in ServerHttpProtocolGenerator and ServerProtocolTestGenerator.
+ */
+class HttpServerTraits {
```

Comment:
I would either:
- have `runtimeConfig` be an argument to this class
- put the methods on the companion object

Reply:
the first one would probably make calling this the cleanest

Reply:
Yeah, I see your point, I'll fix it following 1)